plugins/utils/startup.nix

```diff
@@ -0,0 +1,350 @@
+              (numbers.between 0.0 1.0)
+              ints.positive
+          )
+          "0.5"
+          ''
+            - if < 1, fraction of screen width
+            - if > 1 numbers of column
+          '';
+
+        after = helpers.mkNullOrOption types.str ''
+          A function that gets executed at the end.
+        '';
+
+        emptyLinesBetweenMappings = helpers.defaultNullOpts.mkBool true ''
+          Add an empty line between mapping/commands.
+        '';
+
+        disableStatuslines = helpers.defaultNullOpts.mkBool true ''
+          Disable status-, buffer- and tablines.
+        '';
+
+        paddings = helpers.defaultNullOpts.mkNullable (with types; listOf ints.unsigned) "[]" ''
+          Amount of empty lines before each section (must be equal to amount of sections).
+        '';
+      };
+
+      mappings = {
+        executeCommand = helpers.defaultNullOpts.mkStr "<CR>" ''
+          Keymapping to execute a command.
+        '';
+
+        openFile = helpers.defaultNullOpts.mkStr "o" ''
+          Keymapping to open a file.
+        '';
+
+        openFileSplit = helpers.defaultNullOpts.mkStr "<c-o>" ''
+          Keymapping to open a file in a split.
+        '';
+
+        openSection = helpers.defaultNullOpts.mkStr "<TAB>" ''
+          Keymapping to open a section.
+        '';
+
+        openHelp = helpers.defaultNullOpts.mkStr "?" ''
+          Keymapping to open help.
+        '';
+      };
+
+      colors = {
+        background = helpers.defaultNullOpts.mkStr "#1f2227" ''
+          The background color.
+        '';
+
+        foldedSection = helpers.defaultNullOpts.mkStr "#56b6c2" ''
+          The color of folded sections.
+          This can also be changed with the `StartupFoldedSection` highlight group.
+        '';
+      };
+
+      parts = mkOption {
+        type = with types; listOf str;
+        default = [];
+        description = "List all sections in order.";
+        example = ["section_1" "section_2"];
+      };
+
+      userMappings = mkOption {
+        type = with types; attrsOf str;
+        description = "Add your own mappings as key-command pairs.";
+        default = {};
+        example = {
+          "<leader>ff" = "<cmd>Telescope find_files<CR>";
+          "<leader>lg" = "<cmd>Telescope live_grep<CR>";
+        };
+      };
+    };
+
+  config = mkIf cfg.enable {
+    assertions = let
+      sectionNames = attrNames cfg.sections;
+      numSections = length sectionNames;
+    in [
+      {
+        assertion = (cfg.options.paddings == null) || (length cfg.options.paddings) == numSections;
+        message = ''
+          Nixvim (plugins.startup): Make sure that `plugins.startup.options.paddings` has the same
+          number of elements as there are sections.
+        '';
+      }
+      {
+        assertion =
+          ((length cfg.parts) <= numSections)
+          && (
+            all
+            (sectionName: elem sectionName cfg.parts)
+            sectionNames
```

Comment:
Sorry this one seems off too, I think you need to check `all (part: builtins.hasAttr part cfg.sections) cfg.parts` in order to check that all parts are defined.

And the message should be `You should not have parts that are not defined in 'plugins.startup.sections'`, I think that condition also avoids the need to check the length

Reply:
Done !